deployments/packet-net.nix

```diff
@@ -2,14 +2,43 @@
 
 with (import ../lib.nix);
 let
-  mkPacketNet = hostname: module: extraopts: { config, name, pkgs, resources, ... }: {
-    deployment.targetHost = hostname + ".aws.iohkdev.io";
+  packetSecrets = import ../static/packet.nix;
+  projects = packetSecrets.projects;
+  accessKeyId = packetSecrets.accessKeyId;
+  getPacketKeyPairName = project: "packet-keypair-${project}";
+  mkPacketKeyPair = project: {
+    name = "packet-keypair-${project}";
+    value = {
+      inherit accessKeyId;
+      project = projects.${project}.id;
+    };
+  };
+  mkPacketKeyPairs = projects: pkgs.lib.listToAttrs (map mkPacketKeyPair projects);
+  mkPacketNet = { hostname, module, type ? "demand", facility ? "any", plan ? "c1.small.x86", project ? "infra", extraopts ? {}, ...}: { config, name, pkgs, resources, ... }: let
+    projectId = projects.${project}.id;
+  in {
+    deployment = let
+      deploymentTypes = {
+        demand = {
+          targetHost = hostname + ".aws.iohkdev.io";
+          targetEnv = "packet";
+          packet = {
+            inherit accessKeyId facility plan;
+            keyPair = resources.packetKeyPairs.${getPacketKeyPairName project};
+            project = projectId;
+          };
+        };
+        legacy = {
+          targetHost = hostname + ".aws.iohkdev.io";
+        };
+      };
+    in deploymentTypes.${type};
+
     imports = [ ../modules/common.nix
                 module
-              ] ++
-              # This loads the packet.net-provided .nix files.
-              map (f: builtins.toPath(toString(./.) + "/${hostname}/" + f))
-                  (builtins.attrNames (builtins.readDir (./. + "/${hostname}")));
+              ] ++ (optionals (type == "legacy")
+              (map (f: builtins.toPath(toString(./.) + "/${hostname}/" + f))
+                  (builtins.attrNames (builtins.readDir (./. + "/${hostname}")))));
```

Comment:
you probably dont want to use `toString` or `toPath`
just `./. + "/${hostname}/${f}` should do

Reply:
Updated